runtime/near-vm-runner/src/wasmtime_runner.rs

```diff
@@ -36,37 +37,39 @@ impl WasmtimeMemory {
     }
 }
 
+fn with_caller<T>(func: impl FnOnce(&mut Caller) -> T) -> T {
+    CALLER.with(|caller| func(caller.borrow_mut().as_mut().unwrap()))
+}
+
 impl MemoryLike for WasmtimeMemory {
-    fn fits_memory(&self, offset: u64, len: u64) -> bool {
-        CALLER.with(|caller| match offset.checked_add(len) {
-            None => false,
-            Some(end) => self.0.data_size(caller.borrow_mut().as_mut().unwrap()) as u64 >= end,
-        })
+    fn fits_memory(&self, offset: u64, len: u64) -> Result<(), ()> {
+        let start = usize::try_from(offset).map_err(|_| ())?;
+        let len = usize::try_from(len).map_err(|_| ())?;
+        let end = start.checked_add(len).ok_or(())?;
```

Comment:
This can probably avoid one branch by first adding and then attempting a conversion, but it doesn’t matter much either way as our memories are sized in megabytes.